### PR TITLE
Fixing bugs

### PR DESCRIPTION
**Pull request by bcgoss**
Changed lots of things to work with Feed back received today.

### CODE REVIEW

**Comment by bcgoss** [comment]
@case-eee @s-espinosa We're wondering about to_param. Right now, the URL works with spaces because the spaces get converted to %20. what else would you expect us to handle in this method?

**Comment by s-espinosa** [comment]
Understand that the application may still render the appropriate page with a `%20`replacing a space, but the main idea here is to provide users with vanity links that they can then pass to their friends so they can buy all our stuff even if they don't know how to encode a space in a URL. Recommend you keep working to see how you can parameterize those categories to be used in the link.

**Comment by s-espinosa** [comment]
BTW, I wouldn't necessarily let that hold up merging this PR.

File: app/models/cart.rb [hunk 1]
```diff
@@ -22,9 +22,13 @@ def item_quantities
...
 
+  def empty_message
+    "False Start!  You have nothing in your cart!"
+  end
+
   def total
-    item_quantities.reduce(:+) do |_total, content|
-      content.first.price * content.last
+    item_quantities.reduce(0) do |total, content|
+      total += content.first.price * content.last
```
[the remark this answers is not in the document]
Comment (bcgoss) [reply app/models/cart.rb:31]: This won't work without reduce(0) in the case where there are no items.

**Pull request merged**